### ISSUE TITLE
Make linting actually useful

### ISSUE THREAD

**Issue by sharkwouter** [sharkwouter/minigalaxy#272]
Currently we're using Flake8 for linting the code, but it's kinda pointless as it always passes regardless of the amount of issues in the code. I want to get rid of most of them and make specific linting errors fail the tests, to keep code quality up.

**Comment by Summertime** [comment]
for typechecking, https://github.com/microsoft/pyright, https://github.com/facebook/pyre-check, https://github.com/python/mypy

would need tagging the expected types for functions et al tho

**Comment by viacheslavka** [comment]
Which errors would you like to be "critical"?

On a similar note, is there any policy around stylistic changes? For example, I see code like
```
if something:
    return True
else:
    return False
```
and I want to change it to just
```
return something
```

Would you be OK with this?

**Comment by makson96** [comment]
As far as I know we don't have strict coding rules anywhere. From my side I would prefer, that PyCharm doesn't report any PEP-8 violation.

As it goes to your example - yes I do like single return in function much more, than multiple returns.

**Comment by sharkwouter** [comment]
We do actually have a strict check now. This was fixed in #326

**Comment by sharkwouter** [comment]
I do agree with @viacheslavka though on the return statements.
